state/caasstate.go

```diff
@@ -253,3 +257,35 @@ func (st *CAASState) AddCAASApplication(args AddCAASApplicationArgs) (_ *CAASApp
 	}
 	return app, nil
 }
+
+func (st *CAASState) UpdateUploadedCharm(info CharmInfo) (*Charm, error) {
+	return updateUploadedCharm(st, info)
+}
+
+func (st *CAASState) PrepareLocalCharmUpload(curl *charm.URL) (chosenURL *charm.URL, err error) {
+	return prepareLocalCharmUpload(st, curl)
+}
+
+func (st *CAASState) PrepareStoreCharmUpload(curl *charm.URL) (*Charm, error) {
+	return prepareStoreCharmUpload(st, curl)
+}
+
+// AllCharms returns all charms in state.
+func (st *CAASState) AllCharms() ([]*Charm, error) {
```

Comment:
seems like it would be better to extract the guts of this and call from here and `State.AllCharms`